askup/views.py

```diff
@@ -736,21 +747,48 @@ def answer_evaluate(request, answer_id, evaluation):
     if filter:
         # If it's a Quiz
         filter = get_clean_filter_parameter(request)
-        qset_id = answer.question.qset_id
-        queryset = Question.objects.filter(
-            qset_id=qset_id,
-            vote_value__lte=answer.question.vote_value,
-            text__gt=answer.question.text,
+        previous_question = answer.question
+        qset_id = previous_question.qset_id
+        questions_queryset = QsetView.get_real_questions_queryset(qset_id)
+        questions_queryset = questions_queryset.filter(
+            vote_value__lte=previous_question.vote_value,
         )
-        queryset = apply_filter_to_queryset(request, filter, queryset)
-        next_question = queryset.order_by('-vote_value', 'text').first()
+        questions_queryset = apply_filter_to_queryset(request, filter, questions_queryset)
+        next_question = get_next_quiz_question(questions_queryset, previous_question)
 
         if next_question:
             return get_quiz_question_redirect(next_question.id, filter)
 
     return redirect(reverse('askup:qset', kwargs={'pk': answer.question.qset_id}))
 
 
+def get_next_quiz_question(questions_queryset, previous_question):
+    """
+    Get next quiz Question object from db by the previous one.
+    """
+    for question in questions_queryset:
+        if check_question_is_next(question, previous_question):
+            return question
+
+    return None
+
+
+def check_question_is_next(question, previous_question):
+    """
+    Check if the question is the next one.
+    """
+    if question.id == previous_question.id:
+        return False
+
+    if question.vote_value != previous_question.vote_value:
```

Comment:
@KhaimovMR could you please explaine the meaning of the following nex two statements? They are not clear enough, some comment strings would improve misunderstanding.

Reply:
Rewriting the next question pick functionality for the cache one.